### PR TITLE
optimize to rebalance multiple shards on single rebalance process

### PR DESCRIPTION
**Pull request by abesanderson**
Issue #21163
Optimize to speed up the time to reach shard balance across regions

### CODE REVIEW

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala [hunk 1]
```diff
@@ -149,7 +149,7 @@ object ShardCoordinator {
           case (_, v) ⇒ v.filterNot(s ⇒ rebalanceInProgress(s))
         }.maxBy(_.size)
         if (mostShards.size - leastShards.size >= rebalanceThreshold)
-          Future.successful(Set(mostShards.head))
+          Future.successful(mostShards.take(maxSimultaneousRebalance - rebalanceInProgress.size).toSet)
```
Comment (patriknw) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala]: nice solution!

**Comment by patriknw** [comment]
LGTM
needs a rebase

**Comment by drewhk** [comment]
LGTM

**Comment by typesafehub-validator** [comment]
Hi @abesanderson-vivint,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by abesanderson** [comment]
Hey, I'm not sure why it's prompting for CLA again.  I had some private work projects that were clouding up my email, so I made a second account tied to my work email.  The project in question is not on that account, so, I'm confused what the check is doing.

**Comment by patriknw** [comment]
I have confirmed that you have signed the CLA. It would be convenient if you can sign it again with the other email so that it passes our automatic CLA checks. We can skip that for this PR but if you continue to contribute (which we would like) it will good.

The rebase looks strange. There should only be one commit in the PR. Feel free to close and open new PR if things got messed up beyond recovery.

**Pull request closed without merging**